docs/dev/guidelines/database.rst

```diff
@@ -0,0 +1,236 @@
+**********************
+Database Relationship
+**********************
+
+WORK IN PROGRESS
+
+1. Retrieving and Building Objects
+==================================
+
+The cost of retrieving and building an object's relationships far exceeds the cost of selecting the object. This is especially true for relationships where it would trigger the loading of every child through the relationship hierarchy. The solution to this issue is **lazy fetching** (lazy loading). Lazy fetching allows the fetching of a relationship to be deferred until it is accessed. This is important not only to avoid the database access, but also to avoid the cost of building the objects if they are not needed. |br|
+
+In JPA lazy fetching can be set on any relationship using the fetch attribute. The fetch can be set to either ``LAZY`` or ``EAGER`` as defined in the ``FetchType`` enum. The default fetch type is ``LAZY`` for all relationships except for **OneToOne** and **ManyToOne**, but in general it is a good idea to make every relationship ``LAZY``. The ``EAGER`` default for **OneToOne** and **ManyToOne** is for implementation reasons (more easier to implement), not because it is a good idea. |br|
+
+We **always** use ``FetchType.LAZY``, unless there is a very strong case to be made for ``FetchType.EAGER``.
+
+       .. note::
+        Additional effort to use ``FetchType.LAZY`` does count as a strong argument.
+
+2. Relationships
+================
+
+A relationship is a reference from one object to another. In a relational database relationships are defined through foreign keys. The source row contains the primary key of the target row to define the relationship (and sometimes the inverse). A query must be performed to read the target objects of the relationship using the foreign key and primary key information. In Java, if a relationship is to a collection of other objects, a ``Collection`` or ``array`` type is used in Java to hold the contents of the relationship. In a relational database, collection relations are either defined by the target objects having a foreign key back to the source object's primary key, or by having an intermediate join table to store the relationship (both objects' primary keys). |br|
```

Comment:
In Java, .... in Java - you could leave the second one out just ` a Collection or array type is used to hold...`